idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/Contents.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import org.immutables.value.Value;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Value.Immutable
+public abstract class Contents {
+    private static final Logger log = LoggerFactory.getLogger(Contents.class);
+
+    protected abstract String pageContent();
+
+    public static ImmutableContents pageContents(URL pageUrl) {
+        @Nullable String content = null;
+        ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
+
+        if (indicator == null) {
+            return ImmutableContents.builder().pageContent("").build();
```

Comment:
In general, using "sentinel" values like this is frowned on. It's better to expose the face the operation can fail in the type - in this case this method would return an `Optional<Contents>` which would be `Optional.empty()` if it can't be created for some reason.

Reply:
Thanks for the tip! make alot of the upstream code simpler for checking if it is empty as well